File: serde-generate/runtime/java/com/novi/serde/Serializer.java
// Copyright (c) Facebook, Inc. and its affiliates
// SPDX-License-Identifier: MIT OR Apache-2.0

package com.novi.serde;

import java.lang.Exception;
import java.math.BigInteger;

public interface Serializer {
    void serialize_str(String value) throws Exception;

    void serialize_bytes(Bytes value) throws Exception;

    void serialize_bool(Boolean value) throws Exception;

    void serialize_unit(Unit value) throws Exception;

    void serialize_char(Character value) throws Exception;

    void serialize_f32(Float value) throws Exception;

    void serialize_f64(Double value) throws Exception;

    void serialize_u8(@Unsigned Byte value) throws Exception;

    void serialize_u16(@Unsigned Short value) throws Exception;

    void serialize_u32(@Unsigned Integer value) throws Exception;

    void serialize_u64(@Unsigned Long value) throws Exception;

    void serialize_u128(@Unsigned @Int128 BigInteger value) throws Exception;

    void serialize_i8(Byte value) throws Exception;

    void serialize_i16(Short value) throws Exception;

    void serialize_i32(Integer value) throws Exception;

    void serialize_i64(Long value) throws Exception;

    void serialize_i128(@Int128 BigInteger value) throws Exception;

    void serialize_len(long value) throws Exception;

    void serialize_variant_index(int value) throws Exception;

    void serialize_option_tag(boolean value) throws Exception;

    int get_buffer_offset();

    void sort_map_entries(int[] offsets);

    byte[] get_bytes();
}

File: serde-generate/runtime/swift/Sources/Serde/Serializer.swift
//  Copyright © Diem Association. All rights reserved.

import Foundation

protocol Serializer {
    func serialize_str(value: String) throws
    func serialize_bytes(value: [UInt8])
    func serialize_bool(value: Bool)
    func serialize_unit(value: Unit)
    func serialize_char(value: Character) throws
    func serialize_f32(value: Float)
    func serialize_f64(value: Double)
    func serialize_u8(value: UInt8)
    func serialize_u16(value: UInt16)
    func serialize_u32(value: UInt32)
    func serialize_u64(value: UInt64)
    func serialize_u128(value: BigInt8) throws
    func serialize_i8(value: Int8)
    func serialize_i16(value: Int16)
    func serialize_i32(value: Int32)
    func serialize_i64(value: Int64)
    func serialize_i128(value: BigInt8) throws
    func serialize_len(value: Int64)
    func serialize_variant_index(value: Int)
    func serialize_option_tag(value: Bool)
    func increase_container_depth() throws
    func decrease_container_depth()
    func get_buffer_offset() -> Int
    func sort_map_entries(offsets: [Int])
    func get_bytes() -> [UInt8]
}
